clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -218,7 +226,7 @@ private ProducerBatch createBatchOffAccumulatorForRecord(Record record, int batc
                 record.key(), record.value(), record.headers()), batchSize);
         ByteBuffer buffer = ByteBuffer.allocate(initialSize);
         MemoryRecordsBuilder builder = MemoryRecords.builder(buffer, magic(), recordsBuilder.compressionType(),
-                TimestampType.CREATE_TIME, 0L, recordsBuilder.isTransactional());
+                TimestampType.CREATE_TIME, 0L);
```

Comment:
Just curious, it seems that the transaction is the same as the parent batch before the change. Wouldn't that preserve the transactional flag?